operations/helm/charts/grafana-agent/templates/hpa.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and (eq .Values.controller.type "deployment") .Values.controller.autoscaling.enabled }}
+{{- if and (or (eq .Values.controller.type "deployment") (eq .Values.controller.type "statefulset" ))  .Values.controller.autoscaling.enabled }}
```

Comment:
We could simplify this to `(not eq .Values.controller.type "daemonset")` if you think it's preferred.